src/Components/Endpoints/src/Binding/HttpContextFormDataProvider.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.ObjectModel;
+
+namespace Microsoft.AspNetCore.Components.Binding;
+
+internal class HttpContextFormDataProvider : FormDataProvider, IHostEnvironmentFormDataProvider
```

Comment:
Why does `HttpContextFormDataProvider` implement `IHostEnvironmentFormDataProvider`?

Reply:
Same as AuthenticationStateProvider, it need to be setup in different contexts